### PR TITLE
Added spherical coords functionality to plot_bloch_vector

### LINKED ISSUE #4802: Plotting bloch sphere using spherical coordinates
### What is the expected enhancement?

Functionality to accept Spherical coordinates as well as typical Cartesian for plot_bloch_vector function under state_visualization.py 
(https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/visualization/state_visualization.py)

### PR DESCRIPTION
**Pull request by Vismai-Khanderao**
### Summary

Resolves #4802 

### Details and comments

Optional argument added to plot_bloch_vector which defaults to "cartesian" as to still be functional to previous uses of function, argument can be set to "spherical" if bloch is given with spherical coords.

### CODE REVIEW

**Comment by Vismai-Khanderao** [comment]
To add, would it be neater to instead have the coord_type parameter as a keyword argument instead?

**Comment by nonhermitian** [comment]
So the reason we support only (x,y,z) is because those are directly measurable in a simple manner.  Also these axes correspond to rotation axes for the basic Pauli gates.

**Comment by Vismai-Khanderao** [comment]
I understand, the main uses I see for spherical coordinates is ease of visualisations with spherical coordinates on hand with compilers such as jupyter notebook, but I do see your point as plot_bloch_vector would mostly be used by functions with cartesian coordinates. Hence, the optional parameter at the end.

**Comment by nonhermitian** [comment]
I would also add that mixed states cannot be added unless you add the radius as a coordinate.

**Comment by nonhermitian** [comment]
Oh nevermind. I missed that it was there. Sorry.

File: qiskit/visualization/state_visualization.py [hunk 1]
```diff
@@ -186,17 +186,20 @@ def plot_state_hinton(state, title='', figsize=None, ax_real=None, ax_imag=None,
         return fig
 
 
-def plot_bloch_vector(bloch, title="", ax=None, figsize=None):
+def plot_bloch_vector(bloch, title="", ax=None, figsize=None, *, coord_type="cartesian"):
```
Comment (1ucian0) [inline qiskit/visualization/state_visualization.py:189]: why is the `*` needed?

Comment (Vismai-Khanderao) [reply qiskit/visualization/state_visualization.py:189]: Apologies, I was trying out different ways to accept the argument. The * can be removed.

Comment (Vismai-Khanderao) [reply qiskit/visualization/state_visualization.py]: Fixed it in a recent commit

**Comment by 1ucian0** [comment]
Do you mind adding a release note with an example?

**Comment by Vismai-Khanderao** [comment]
@1ucian0 Added the note with an example, let me know if it needs any edits!

File: releasenotes/notes/spherical-coord-bloch-295beddb37ef16a6.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+---
+features:
+  - |
+    Added a feature for bloch vectors to be plotted using spherical coordinates.
+    Implements feature mentioned in issue `#4802 <https://github.com/Qiskit/qiskit-terra/issues/4802>`
+
+    Cartesian coordinates are still supported and made as the default mode.
+    for example::
+
+      from qiskit.visualization import plot_bloch_vector
```
Comment (1ucian0) [inline releasenotes/notes/spherical-coord-bloch-295beddb37ef16a6.yaml:10]:
```suggestion
    for example::
    
    .. jupyter-execute::

      from qiskit.visualization import plot_bloch_vector
```

**Review by 1ucian0 — APPROVED** [review]
Thanks a lot and welcome to the community!

**Pull request merged**